launch/launch/actions/execute_process.py

```diff
@@ -314,18 +317,52 @@ def __on_process_stdin(
     def __on_process_stdout(
         self, event: ProcessIO
     ) -> Optional[SomeActionsType]:
-        for line in event.text.decode(errors='replace').splitlines():
-            self.__stdout_logger.info(
-                self.__output_format.format(line=line, this=self)
-            )
+        self.__stdout_buffer.write(event.text.decode(errors='replace'))
+        self.__stdout_buffer.seek(0)
+        last_line = None
+        for line in self.__stdout_buffer:
+            if line.endswith(os.linesep):
+                self.__stdout_logger.info(
+                    self.__output_format.format(line=line[:-len(os.linesep)], this=self)
+                )
+            else:
+                last_line = line
+                break
+        self.__stdout_buffer.truncate(0)
+        if last_line is not None:
+            self.__stdout_buffer.write(last_line)
 
     def __on_process_stderr(
         self, event: ProcessIO
     ) -> Optional[SomeActionsType]:
-        for line in event.text.decode(errors='replace').splitlines():
-            self.__stderr_logger.info(
-                self.__output_format.format(line=line, this=self)
-            )
+        self.__stderr_buffer.write(event.text.decode(errors='replace'))
+        self.__stderr_buffer.seek(0)
+        last_line = None
+        for line in self.__stderr_buffer:
+            if line.endswith(os.linesep):
+                self.__stderr_logger.info(
+                    self.__output_format.format(line=line[:-len(os.linesep)], this=self)
+                )
+            else:
+                last_line = line
+                break
+        self.__stderr_buffer.truncate(0)
+        if last_line is not None:
+            self.__stderr_buffer.write(last_line)
+
+    def __flush_buffers(self, event, context):
+        with self.__stdout_buffer as buf:
+            line = buf.getvalue()
+            if line != '':
+                self.__stdout_logger.info(
+                    self.__output_format.format(line=line, this=self)
+                )
+        with self.__stderr_buffer as buf:
+            line = buf.getvalue()
+            if line != '':
+                self.__stdout_logger.info(
```

Comment:
Shouldn't this be logged on `__stderr_logger`?

Reply:
True that.